PPC64 FIT loading

krystian-hebel opened the pull request.

krystian-hebel committed:
arch/ppc64: initial development towards FIT payloads

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I91cdebf1135442a92a724eb864413dad8103fc92

krystian-hebel committed:
lib/fit_payload: comment out call to fit_update_memory()

Temporary comment out this call, it removes information about memory
associativity. Revert it later.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

IgorBagnucki commented on src/soc/ibm/power9/rom_media.c:
```diff
@@ -14,7 +14,7 @@
 #define LPC_FLASH_TOP (LPC_FLASH_MIN + FW_SPACE_SIZE)
 
 /* TODO: at some point we may have to change it to a bigger partition (HBI?) */
```
If the partition is changed, please remove the TODO

krystian-hebel committed:
lib/fit_payload.c: move call to dt_apply_fixups()

This patch moves a call to dt_apply_fixups() further into FDT preparation.
It allows the fixups to modify 'chosen' and 'memory' nodes, which weren't
yet present when dt_apply_fixups() was called earlier.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I13caa4e033bef07eaab33516d087de642e10e093

Commit:
Fit payload building (#67)

* Prepare files for building FIT payload

* payloads/external/skiboot: include choice in menu

* payloads/external/skiboot: Add building of uImage

* payloads/external/skiboot: Include skiboot in build

* Select correct compression algorithm

* Change ROM size to 2048 and fix skiboot cleaning

* Fixes after review part 1

* Add cleaning of fdt.bin

* Add board-specific options to Kconfig

* Build skiboot payload intermediate files inside build directory

* Fix skiboot clean

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
soc/ibm/power9: add required "ibm,chip-id" property to the device tree

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: Ic45fae90350a968f0a1c813d673c139f58db21e3

krystian-hebel committed:
soc/ibm/power9: fill memory table with real values

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I52019bba54cb62134d9409c47363f7604d8e130a

krystian-hebel committed:
mainboard/talos-2: shorten DTS file name, remove memory nodes

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I0bdac7ca3125fe299c6f3fae56fcd9034b567f1c

krystian-hebel committed:
soc/power9/rom_media.c: remove obsolete comment

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I1d6e6c8467a83ec77a0da4e02a786e4ac08a28de

macpijan committed:
.github/workflows/build.yml: use 3mdeb/coreboot-sdk

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
